File: CMPS12BM Winter19/lab2/FileReverse.java
import java.io.*;
import java.util.Scanner;

class FileReverse{
	public static void main(String[] args) throws IOException{
		Scanner in = null;
		PrintWriter out = null;
		String line = null;
		String[] token = null;
		int i, n, lineNumber = 0;
		// check number of command line arguments is at least 2
		if(args.length < 2){
			System.out.println("Usage: FileCopy <input file> <output file>");
			System.exit(1);
		}
		// open files
		in = new Scanner(new File(args[0]));
		out = new PrintWriter(new 	FileWriter(args[1]));
		// read lines from in, write lines to out
		while( in.hasNextLine() ){
			lineNumber++;
			line = in.nextLine().trim() + " ";
			token = line.split("\\s+");
			n = token.length;
			//out.println( line ); This accidentally reposts the initial input
			for(i = 0; i < n; i++){
				out.println(stringReverse(token[i], token[i].length()));
			}
		}
		// close files
		in.close();
		out.close();
	}
	//I'm a goof and it took me forever to figure out the recursive method
	public static String stringReverse(String s, int n){
		if(s.length() == 1){
			return s;
		}
		else{ 
			return stringReverse(s.substring(1),n-1) + s.charAt(0);
		}
	}
}

File: CS12BM Spring19/lab3/FileReverse.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>

void stringReverse(char* w){
        int i = 0, j = strlen(w)-1;
        while(i<j){
                char temp = w[i];
                w[i] = w[j];
                w[j] = temp;
                i++;
                j--;
        }
}
int main(int argc, char * argv[]){
        FILE *in, *out;//handles for input and output files
        char word[256];//char array to store words from input file

        //check command line for correct number of arguments
        if(argc != 3){
                printf("Usage: %s <input file> <output file>\n", argv[0]);
                exit(EXIT_FAILURE);
        }

        //open input file for reading
        in = fopen(argv[1], "r");//"r" for reading
        if(in == NULL){
                printf("Unable to read from file %s\n", argv[1]);
                exit(EXIT_FAILURE);
        }

        //open output file for writing
        out = fopen(argv[2], "w");//"w" for writing
        if(out == NULL){
                printf("Unable to write to file %s\n", argv[2]);
                exit(EXIT_FAILURE);
        }

        //read words from input file, print on separate lines to output file
        while(fscanf(in, " %s", word) != EOF){ //EOP = End of File
                stringReverse(word);
                fprintf(out, "%s\n", word);
        }

        //close input and output files
        fclose(in);
        fclose(out);
        return(EXIT_SUCCESS);
}